framework/src/constraints/AutomaticMortarGeneration.C

```diff
@@ -568,11 +585,43 @@ AutomaticMortarGeneration::buildMortarSegmentMesh()
     mortar_segment_set.insert(new_elem_right);
   }
 
-  // Set up the the mortar segment neighbor information.
-  mortar_segment_mesh->allow_renumbering(true);
-  mortar_segment_mesh->skip_partitioning(true);
-  mortar_segment_mesh->allow_find_neighbors(true);
-  mortar_segment_mesh->prepare_for_use();
```

Comment:
Sorry @recuero, didn't see this till just now (didn't expand this file). Yeah essentially I removed them because I thought they were unnecessary since we're using this mesh as more of a holder for mortar segments but it probably doesn't save much. I'll put them back in.

Reply:
Added back in to both the 2D and 3D mortar segment mesh generation